src/System/IO/Resource/Linear/Internal.hs

```diff
@@ -147,6 +158,22 @@ hPutStrLn h s = flipHPutStrLn s h -- needs a multiplicity polymorphic flip
     flipHPutStrLn s =
       coerce (unsafeFromSystemIOResource_ (\h' -> Text.hPutStrLn h' s))
 
+-- | See @System.IO.'System.IO.hSeek'@.
+--
+-- @since 0.2.1
+hSeek :: Handle %1 -> System.SeekMode -> Integer -> RIO Handle
+hSeek h mode i = coerce hSeek' h
+  where
+    hSeek' :: Handle %1 -> RIO Handle
+    hSeek' =
+      coerce (unsafeFromSystemIOResource_ (\h' -> System.hSeek h' mode i))
+
+-- | See @System.IO.'System.IO.hTell'@.
+--
+-- @since 0.2.1
+hTell :: Handle %1 -> RIO (Ur Integer, Handle)
+hTell = coerce (unsafeFromSystemIOResource System.hTell)
```

Comment:
It is not clear to me why this needs to consume the handle; maybe `Handle -> RIO (Ur Integer)` is fine since we're only querying? But I cribbed the type from `hIsEOF` where I'd expect similar reasoning to apply.

Reply:
You say "keep in mind" as if that was a thing that had ever entered my mind :-)

I don't understand you when you say "as we have a linear `Handle` in scope, we can't ask for an unrestricted copy of it." I thought linearity in this extension was a property of function arrows, not of values?

What does make sense to me is that I just realised `RIO` only has a linear control monad instance, which means we couldn't compose with a function of type `Handle -> RIO (Ur Integer)`, which means we have to use a linear arrow, which means we have to be careful to return the `Handle`. What other parts of the picture am I missing?

Reply:
> You say "keep in mind" as if that was a thing that had ever entered my mind :-)

:slightly_smiling_face: 

> I thought linearity in this extension was a property of function arrows, not of values?

The terminology is a little ambiguous here. A “linear function” is a function with “linear argument” (it's the argument, the formal parameter, that is linear not its value).

> What does make sense to me is that I just realised `RIO` only has a linear control monad instance, which means we couldn't compose with a function of type `Handle -> RIO (Ur Integer)`, which means we have to use a linear arrow, which means we have to be careful to return the `Handle`. What other parts of the picture am I missing?

Yes, this is exactly what's happening.